test/SConstruct

```diff
@@ -27,28 +28,59 @@ else:
 
 env = Environment(ENV=os.environ)
 
+is64 = sys.maxsize > 2 ** 32
+if (
+    env["TARGET_ARCH"] == "amd64"
+    or env["TARGET_ARCH"] == "emt64"
+    or env["TARGET_ARCH"] == "x86_64"
+    or env["TARGET_ARCH"] == "arm64-v8a"
+):
+    is64 = True
+
 opts = Variables([], ARGUMENTS)
 
 # Define our options
-opts.Add(EnumVariable("target", "Compilation target", "debug", allowed_values=("debug", "release"), ignorecase=2))
 opts.Add(
     EnumVariable(
         "platform",
-        "Compilation platform",
+        "Target platform",
         host_platform,
-        # We'll need to support these in due times
-        # allowed_values=("linux", "freebsd", "osx", "windows", "android", "ios", "javascript"),
-        allowed_values=("linux", "windows"),
+        allowed_values=("linux", "freebsd", "osx", "windows", "android", "ios", "javascript"),
         ignorecase=2,
     )
 )
-opts.Add(EnumVariable("bits", "Target platform bits", "64", ("32", "64")))
-opts.Add(BoolVariable("use_llvm", "Use the LLVM / Clang compiler", "no"))
-opts.Add(PathVariable("target_path", "The path where the lib is installed.", default_target_path, PathVariable.PathAccept))
+opts.Add(EnumVariable("bits", "Target platform bits", "64" if is64 else "32", ("32", "64")))
+opts.Add(BoolVariable("use_llvm", "Use the LLVM compiler - only effective when targeting Linux or FreeBSD", False))
+opts.Add(BoolVariable("use_mingw", "Use the MinGW compiler instead of MSVC - only effective on Windows", False))
+# Must be the same setting as used for cpp_bindings
+opts.Add(EnumVariable("target", "Compilation target", "debug", allowed_values=("debug", "release"), ignorecase=2))
+opts.Add(
+    PathVariable("target_path", "The path where the lib is installed.", default_target_path, PathVariable.PathAccept)
+)
 opts.Add(PathVariable("target_name", "The library name.", default_library_name, PathVariable.PathAccept))
```

Comment:
Worth a discussion whether we actually want target_name as a path variable, I have yet to find a good use case where you would change this :) I think it was added way back when we first started with GDNative more out of habit than for any real reason.